src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrPatchHandler.cs

```diff
@@ -608,20 +608,74 @@ static string SanitizeName(string mappingName)
             cancellationToken);
     }
 
-    public IReadOnlyCollection<string> GetVmrPatches(string mappingName)
+    /// <summary>
+    /// Returns paths to all VMR patches.
+    /// </summary>
+    /// <param name="patchVersion">Version of the VMR patches to get or null if to take current</param>
+    /// <returns>List of VMR patches belonging to the given mapping</returns>
+    public async Task<IReadOnlyCollection<VmrIngestionPatch>> GetVmrPatches(
+        string? patchVersion,
+        CancellationToken cancellationToken)
+        => patchVersion == null
+            ? GetWorkingTreeVmrPatches()
+            : await GetVmrPatchesFromHistory(patchVersion, cancellationToken);
+
+    private IReadOnlyCollection<VmrIngestionPatch> GetWorkingTreeVmrPatches()
+    {
+        if (_vmrInfo.PatchesPath is null)
+        {
+            return [];
+        }
+
+        var patchesDir = _vmrInfo.VmrPath / _vmrInfo.PatchesPath;
+
+        if (!_fileSystem.DirectoryExists(patchesDir))
+        {
+            return [];
+        }
+
+        return _fileSystem
+            .GetDirectories(_vmrInfo.VmrPath / _vmrInfo.PatchesPath)
+            .Select(dir => (Mapping: _fileSystem.GetFileName(dir)!, Patches: _fileSystem.GetFiles(dir)))
+            .SelectMany(pair => pair.Patches.Select(patch => new VmrIngestionPatch(patch, VmrInfo.SourcesDir / pair.Mapping)))
+            .ToArray();
+    }
+
+    private async Task<IReadOnlyCollection<VmrIngestionPatch>> GetVmrPatchesFromHistory(
+        string patchVersion,
+        CancellationToken cancellationToken)
     {
         if (_vmrInfo.PatchesPath is null)
         {
-            return Array.Empty<string>();
+            return [];
+        }
+
+        var result = await _localGitClient.RunGitCommandAsync(
+            _vmrInfo.VmrPath,
+            ["ls-tree", "-r", "--name-only", patchVersion, "--", _vmrInfo.PatchesPath],
+            cancellationToken);
+
+        if (!result.Succeeded)
```

Comment:
Is it fine to just quietly return an empty list for any failure of the command?

Reply:
Yeah, it might mean the file was not existent previously which means it is getting added (or vice versa).